blog/2024-09-17-better-ecommerce-typescript/index.mdx

```diff
@@ -0,0 +1,507 @@
+---
+title: AI Enabled eCommerce with Weaviate and TypeScript
+slug: ai-enabled-ecommerce-typescript
+authors: [daniel]
+date: 2024-09-17
+tags: ['concepts', 'how-to', 'search']
+image: ./img/hero.png
+description: 'We dive into how AI can craft better eCommerce experiences with a focus on one critical component; Search.'
+---
+
+![AI Enabled eCommerce with Weaviate and TypeScript](./img/hero.png)
+
+Today, there are [2.64 billion online buyers](https://www.yaguara.co/online-shopping-statistics/#:~:text=online%20shopping%2Dconsumer.-,There%20are%202.64%20billion%20online%20buyers%20in%20the%20world%20as,3.1%25%20increase%20in%20online%20buyers.) in the world. Making up more than 33% of the world’s total population. For comparison, [5.35 billion people](https://www.forbes.com/home-improvement/internet/internet-statistics/#:~:text=There%20are%205.35%20billion%20internet%20users%20worldwide.&text=Out%20of%20the%20nearly%208,the%20internet%2C%20according%20to%20Statista.) have access to the internet as of 2024. Almost half of the people using the internet shop online. However, that experience could be anything but perfect. A [Google report ](https://cloud.google.com/blog/topics/retail/search-abandonment-impacts-retail-sales-brand-loyalty)shows that nearly $300 billion is lost each year from bad online search experiences with 85% of global online customers viewing a brand differently after an unsuccessful search. Search clearly very important for great e-commerce experiences. 
+
+
+Here are a few more stats to drive the point home.
+
+
+* Over [50% of eCommerce sales](https://www.doofinder.com/en/blog/ecommerce-site-search-statistics) can be traced back to customers who use the search bar! 
+* It turns out that [34% of online shoppers](https://www.doofinder.com/en/blog/non-product-searches) try to search for non-product content. 
+* [87% of shoppers begin](https://www.salesforce.com/blog/2018/08/digital-shopper-first-retail-report-research.html) their product searches online.
+* 43% of users on retail websites [go directly to the search bar](https://www.forrester.com/report/MustHave+eCommerce+Features/-/E-RES89561).
+* Site searchers are [2-3x more likely to convert.](https://neilpatel.com/blog/site-search-killing-your-conversion/)
+
+A few standouts, but clearly, all drive the conclusion that search is a critical component of any e-commerce product.  
```

Comment:
Not sure how to best incorporate this, but maybe you have an idea: It would be great to have a heading titled "How AI is used in e-commerce". Maybe somewhere around here?

If you don't think it makes sense, feel free to ignore.